src/main/scala/com/advancedtelematic/campaigner/http/CampaignResource.scala

```diff
@@ -58,7 +59,8 @@ class CampaignResource(extractAuth: Directive1[AuthedNamespaceScope],
             complete(campaigns.allCampaigns(ns, offset, limit))
           } ~
           (post & entity(as[CreateCampaign])) { request =>
-            complete(StatusCodes.Created -> createCampaign(ns, request))
```

Comment:
Can you try using NonEmptyList CreateCampaign ? I think you might get this for free from circe.

Needs rebase

Reply:
Alright, so in `CreateCampaign` I tried turning the `Set[GrouId]` to a `NonEmptySet[GroupId]`, but the decoders don't work anymore and I can't find a way without implementing the whole encoder/decoder.
Any tips?

Reply:
You just need a decoder for `NonEmptySet[T]` given there is a `Decoder[T]`. Me or Simao can help you with it if you don't know how. It is quite simple

Reply:
I was almost sure circe came with that decoder though

Reply:
```
def nesDecoder[T](implicit dec: Decoder[Set[T]]): Decoder[NonEmptySet[T]] =
    dec.emapTry(xs => if(xs.isEmpty) Failure(DecodingFailure("No elements")) else Success(xs))
```

Reply:
Needs NonEmptySet.fromSetUnsafe() or unpack option from NonEmptySet.fromSet() ;)

Reply:
Actually, it should be 
```
    def nesDecoder[T](implicit dec: Decoder[Set[T]], ord: Ordering[T]): Decoder[NonEmptySet[T]] = {
    import scala.collection.immutable.SortedSet
    dec.emapTry(xs => if(xs.isEmpty) Failure(DecodingFailure("No elements", Nil)) else Success(NonEmptySet.fromSetUnsafe(xs.to[SortedSet])))
  }
```

Reply:
let's open a circe pr :+1:

Reply:
We could, but will need to do bit more work to get rid of `Nil` in `DecodingFailure("No elements", Nil)`. And to go through internal OSS contribution process :)

Reply:
I'll create this codecs in libats.

Reply:
Alright, in the end I refactored this to use `NonEmptyList`. WDYT?

Reply:
I think `Set` still makes sense. It does not make sense to have a group several times in this collection

Reply:
I didn't change `CampaignRepository#persist`, that still takes a `Set[GroupId]`.